.github/workflows/build.yml

```diff
@@ -392,6 +418,37 @@ jobs:
         run: cargo add --dev --features bindgen 'aws-lc-sys@>0.20' --package portable-rustls --verbose && cargo install bindgen-cli --verbose
       - run: cross test --package portable-rustls --target ${{ matrix.target }}
 
+  cross-build:
+    name: portable cross build testing
+    runs-on: ubuntu-latest
+    if: github.event_name != 'merge_group'
+    strategy:
+      matrix:
+        target:
+          # 32-bit target(s) with no atomic ptr:
+          # - thumbv6m-none-eabi
+          - riscv32imc-esp-espidf
+          # XXX TBD OTHER TARGET(S) - ???
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+
+      - name: Install rust-toolchain - nightly # Rust nightly needed to test with unstable critical-section feature
+        uses: dtolnay/rust-toolchain@nightly
+      - name: Install cross (cross-rs) from GitHub
+        run: cargo install cross --git https://github.com/cross-rs/cross
+      # XXX TBD add feature to configure critical-section - ???
+      - run: cargo add once_cell -F critical-section -p portable-rustls
+      - run: cargo add portable-atomic -F critical-section -p portable-rustls
+      - run: cargo add spin -F portable_atomic -p portable-rustls
```

Comment:
It took me a little while to figure out that using "portable-atomic" feature with a dash (`-`) didn't work due to https://github.com/mvdnes/spin-rs/issues/170 - proposed https://github.com/mvdnes/spin-rs/pull/172 to fix this.